DataFormats/HcalIsolatedTrack/src/IsolatedPixelTrackCandidate.cc

```diff
@@ -34,33 +34,31 @@ bool IsolatedPixelTrackCandidate::overlap( const Candidate & c ) const {
   return ( o != 0 &&  checkOverlap( track(), o->track() ) );
 }
 
-std::pair<int,int> IsolatedPixelTrackCandidate::towerIndex() const
-{
-  int ieta=0, iphi=0;
-  for (int i=1; i<21; i++)
-    {
-      if (fabs(track_->eta())<(i*0.087)&&fabs(track_->eta())>(i-1)*0.087) ieta=int(fabs(track_->eta())/track_->eta())*i;
-    }
-  if (fabs(track_->eta())>1.740&&fabs(track_->eta())<1.830) ieta=int(fabs(track_->eta())/track_->eta())*21;
-  if (fabs(track_->eta())>1.830&&fabs(track_->eta())<1.930) ieta=int(fabs(track_->eta())/track_->eta())*22;
-  if (fabs(track_->eta())>1.930&&fabs(track_->eta())<2.043) ieta=int(fabs(track_->eta())/track_->eta())*23;
+std::pair<int,int> IsolatedPixelTrackCandidate::towerIndex() const {
 
-  double delta=track_->phi()+0.174532925;
-  if (delta<0) delta=delta+2*acos(-1);
-  if (fabs(track_->eta())<1.740) 
-    {
-      for (int i=0; i<72; i++)
-	{
-	  if (delta<(i+1)*0.087266462&&delta>i*0.087266462) iphi=i;
-	}
+  int ieta(0), iphi(0), nphi(72), kphi(1);
+  double etas[24]={0.000,0.087,0.174,0.261,0.348,0.435,0.522,0.609,0.696,
+		   0.783,0.870,0.957,1.044,1.131,1.218,1.305,1.392,1.479,
+		   1.566,1.653,1.740,1.830,1.930,2.043};
```

Comment:
Please do not duplicate these values of the calo towers boundaries. Include the file RecoJets/JetAnalyzers/interface/CaloTowerBoundries.h to get these eta values.